### ISSUE TITLE
no blank node in object position

### ISSUE THREAD

**Issue by mathib** [RMLio/rmlmapper-java#43]
Hi! I've just used the RML-mapper v4.5.1 and I've issues with the creation of blank nodes. I'm able to successfully create a BN in a subject position, but not in the object position of a triple:

```
rr:subjectMap [
	rr:template "mybn{F41}" ;
	rr:termType rr:BlankNode ;
	rr:class arp:Photograph
]
...
```
returns a BN (in Turtle output file): `_:mybnABC`

```
...
rr:predicateObjectMap [
	rr:predicate vann:example ;
	rr:objectMap [
		rr:template "mybn{F41}" ;
		rr:termType rr:BlankNode
	]
]
```
does not return a BN but an IRI (in Turtle output file): `<mybnABC> `.

My goal is to create output triples as follows:
```
ex:subj vann:example _:bn1 .
_:bn1 a arp:Photograph ;
      rdfs:label "some text" ;
      arp:hasFullFileName "http://ex.org/photo.jpg" .
```

**Comment by pheyvaer** [comment]
Hi @mathib 

This is indeed a bug in the RMLMapper. We will have a look into this. Regarding your specific use case, it might be better to use [Parent Triples Map](https://www.w3.org/TR/r2rml/#dfn-parent-triples-map) instead of generating the blank nodes yourself. That way the RMLMapper will generate a blank node and make sure that the different entities are correctly linked to each other via the blank node.

**Comment by mathib** [comment]
Hi! Thanks for the reply! I briefly looked into Parent Triples Map already, but I had the feeling it should be used in cases when you have multiple source files (or tables in the case of a relational db)? In my case, I'm only considering a single TSV file. But I've just tested this approach and it works for my case (albeit a bit verbose) :)

P.S. I've found that the examples of Parent Triples Map in http://rml.io/spec.html#integrated-mapping contain some syntactical errors (there's twice a set of square brackets too much around the `rr:joinCondition`)
